### PR TITLE
Refresh README

### LINKED ISSUE #1206: Use proper semantic versioning
v 1.10.0 renamed a package (`encoding/x/protobuf`), which means that some components cannot build any longer. `1.10.0` looks like a semantic version, so is there any chance you could keep to non-breaking changes on minor version bumps?

### PR DESCRIPTION
**Pull request by breerly**
Updating with the idea that the README should:

* Only contain a pitch
* Be small (roughly less than a page)
* Make it easy to locate critical info like the Stability section
* Delegate detailed usage docs to Godocs
* Contain badges reflecting master (what customers use)

Resolves #1206

https://github.com/yarpc/yarpc-go/blob/readme-refresh/README.md

### CODE REVIEW

**Comment by breerly** [comment]
Ok, this is ready.

I'm going to take a pass through GoDoc tomorrow to make it easily navigable. The changes for that will come in another PR - digging into the best way to write GoDoc now.

**Comment by peter-edge** [comment]
https://github.com/yarpc/yarpc-go/blob/readme-refresh/README.md

File: README.md [hunk 1]
```diff
@@ -1,297 +1,51 @@
...
-apply to requests over any transport.
-
-Each YARPC transport protocol can implement inbound handlers and outbound
-callers. Each of these can support different RPC types, like unary (request and
-response) or oneway (request and receipt) RPC. A future release of YARPC will
-add support for other RPC types including variations on streaming and pubsub.
+High-level features are implemented *above the wire*, enabling:
 
+* Exposing a server over multiple transports simultaneously.
+* Migrating outbound calls between transports with no code changes using config.
```
Comment (abhinav) [inline README.md]: nit: "no code changes using config" isn't right

> [..] with just config changes.

File: README.md [hunk 2]
```diff
@@ -1,297 +1,51 @@
...
+* Middleware that work even as the encoding and transport vary.
```
Comment (abhinav) [inline README.md]: > Middlewares that work

Or

> Middleware that works

File: README.md [hunk 3]
```diff
@@ -1,297 +1,51 @@
...
-You can then either use `yab` or `curl` to list available procedures:
+No breaking changes will be made to exported APIs before `v2.0.0` with the
+**exception of experimental packages**.
 
-```bash
-$ yab -e json -p "http://${host}:${port}" ${service} yarpc::procedures
-```
+Experimental packages reside within packages named `x`, and are *not stable*. This means their
+APIs can break at any time. The intention here is to iterate closely with internal
+customers validate and mature the contained APIs. Once stable, their contents will be moved out of
```
Comment (abhinav) [inline README.md]: > [..] to validate these APIs and iterate on them by working closely with internal customers.

**Comment by breerly** [comment]
I'm going to give Kris room to block/stamp since he wrote most of the original README.

**Comment by breerly** [comment]
Treating @kriskowal's silence as golden.

**Comment by Willyham** [comment]
@breerly Not a fan of this :cry: 

I understand and agree with the principles that you lay out, but this readme was the only place that a lot of things were documented. You have to consider a lot of the issues which are 'solved' internally that other people are going to want to do.

For example, with this README and the godocs, please:

- Tell me how I can generate YARPC compatible code from Thrift files
- How to call a YARPC service from the command line.
- How to call a YARPC service with a Thrift client
- How to configure a thrift inbound
- How to configure ANY outbound

These are absolutely critical. The old README was messy but it at least answered these questions either directly or indirectly.

It's pretty much impossible to get started with YARPC with the current state of the docs. Can we add back an 'examples' or 'getting started' section that's readable in github?

**Comment by breerly** [comment]
@Willyham thanks for the concern and patience as we work through improving our docs.

I'm eying fixing alot of these issues in #1234, and @peter-edge has suggested similar things to you - surfacing key examples and docs from the README, at least in the form of links.

Give me a few days to get us in a better state.

In the meantime, here is the old [README](https://github.com/yarpc/yarpc-go/blob/5722c215c40a8972dd1e703d8ff2c3a611c46e67/README.md), and I'm going to send you an internal doc that *should* answer alot of those questions.

**Comment by kriskowal** [comment]
This change should have been landed after landing the new documentation. The README needs to link through to all the information that was present here.

Baring that, we should revert or land #1234 with links README here ASAP.

**Comment by breerly** [comment]
I disagree *and* gave you the opportunity to review #1234.

**Comment by breerly** [comment]
@kriskowal I gave you a whole day to review this PR, and you didnt.

I'm not willing to rollback the change - the previous README was all over the place and did more harm than good. Thankfully, we have version control and can still pull up the old one in the short interim while I bring back what's necessary.

**Comment by breerly** [comment]
@Willyham @kriskowal follow along the https://github.com/yarpc/yarpc-go/pull/1241

You'll see that where I'm going - a "Usage" section in the README that links off to several in-depth guides:

screen shot 2017-07-27 at 7 57 30 am

These will cover all the questions @Willyham posted, *and* will consolidate our docs into an easy to find and pleasant structure.

**Comment by Willyham** [comment]
Looks great. Looking forward to this being finished!

**Comment by Willyham** [comment]
Also, if I know enough about yarpc to divide and conquer (questionable) then I'm happy to help with some of the sections. Just let me know.

**Pull request merged**